e2e/authorization_test.go

```diff
@@ -23,135 +23,207 @@ import (
 
 	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/deploy"
 	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/kube"
+	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator"
 	monitoringv1 "github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/apis/monitoring/v1"
 	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/generated/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/client-go/kubernetes"
 )
 
+const errCertificate = "x509: certificate signed by unknown authority"
+const errUnauthorized = "server returned HTTP status 401 Unauthorized"
+const errInvalidClientCredentials = "oauth2: \"invalid_client\" \"incorrect client credentials\""
+
 func TestTLSPodMonitoring(t *testing.T) {
 	ctx := context.Background()
 	kubeClient, opClient, err := setupCluster(ctx, t)
 	if err != nil {
 		t.Fatalf("error instantiating clients. err: %s", err)
 	}
 
-	t.Run("collector-deployed", testCollectorDeployed(ctx, kubeClient))
-	t.Run("enable-target-status", testEnableTargetStatus(ctx, opClient))
-	t.Run("patch-example-app-args", testPatchExampleAppArgs(ctx, kubeClient, []string{"--tls-create-self-signed=true"}))
+	t.Run("patch-example-app-args", testPatchExampleAppArgs(ctx, kubeClient, []string{
+		"--tls-create-self-signed=true",
+	}))
+	authorizationPodMonitoringTest(ctx, t, kubeClient, opClient, "tls",
+		&monitoringv1.ScrapeEndpoint{
+			Scheme:   "https",
+			Port:     intstr.FromString("web"),
+			Interval: "5s",
+			HTTPClientConfig: monitoringv1.HTTPClientConfig{
+				TLS: &monitoringv1.TLS{
+					InsecureSkipVerify: true,
+				},
+			},
+		}, &monitoringv1.ScrapeEndpoint{
+			Scheme:   "https",
+			Port:     intstr.FromString("web"),
+			Interval: "5s",
+		},
+		errCertificate,
+	)
+}
 
-	pm := &monitoringv1.PodMonitoring{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "tls-ready",
-			Namespace: "default",
-		},
-		Spec: monitoringv1.PodMonitoringSpec{
-			Selector: metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"app": "go-synthetic",
-				},
-			},
-			Endpoints: []monitoringv1.ScrapeEndpoint{
-				{
-					Scheme:   "https",
-					Port:     intstr.FromString("web"),
-					Interval: "5s",
-					HTTPClientConfig: monitoringv1.HTTPClientConfig{
-						TLS: &monitoringv1.TLS{
-							InsecureSkipVerify: true,
-						},
-					},
+func TestTLSClusterPodMonitoring(t *testing.T) {
+	ctx := context.Background()
+	kubeClient, opClient, err := setupCluster(ctx, t)
+	if err != nil {
+		t.Fatalf("error instantiating clients. err: %s", err)
+	}
+
+	t.Run("patch-example-app-args", testPatchExampleAppArgs(ctx, kubeClient, []string{
+		"--tls-create-self-signed=true",
+	}))
+	authorizationClusterPodMonitoringTest(ctx, t, kubeClient, opClient, "tls",
```

Comment:
nit(non-blocking): There might be opportunity to make those table tests at some point: `TestTLSPodMonitoring` and `TestTLSClusterPodMonitoring` is only different by one thing - first is invoking `authorizationClusterPodMonitoringTest` second `authorizationPodMonitoringTest`

Reply:
Similar to other tests for diff security types.

Reply:
Another argument for table test is authorizationPodMonitoringTest (and cluster one) has a few arguments where you really have understand order and meaning of them. Table test would allow describing scenarios in explicit struct perhaps.